authentication/allowed_hosts_validator.go

```diff
@@ -36,14 +49,24 @@ func (v *AllowedHostsValidator) SetAllowedHosts(hosts []string) {
 	}
 }
 
+// SetAllowedHostsErrorCheck sets the list of valid hosts with error checking.
+func (v *AllowedHostsValidator) SetAllowedHostsErrorCheck(hosts []string) error {
+	v.validHosts = make(map[string]bool, len(hosts))
+	if len(hosts) > 0 {
+		for _, host := range hosts {
+			if strings.HasPrefix(host, "http://") || strings.HasPrefix(host, "https://") {

```

Comment:
we should normalize host to lower first.

Reply:
I will fix it.

Reply:
We should add some these test cases. When I run go test, they passed.